bedrock/settings/__init__.py

```diff
@@ -125,6 +122,8 @@
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["report-uri"] = csp_ro_report_uri
 
     # CSP directive updates we're testing that we hope to move to the enforced policy.
+    CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["default-src"] = [csp.constants.SELF]
```

Comment:
Basically it's a kitchen sink right now, making sure any service or hotlink would work without thinking about it too much…  Ideally every policy should enumerate the hosts it really needs, e.g. https://github.com/mozilla/bedrock/issues/11943#issuecomment-2255722632 — so that it's restricted to more specific hostnames (like some `services.mozilla.com` for `connect-src`, `assets.mozilla.net` or `cdn.mozilla.net` for wherever it's needed etc…)

Which should be the next move in tightening, though. Along with specifically allowing just self/www.m.o where needed, instead of *.m.o that would allow anything (XSS) from e.g. MDN, addons.m.o, discourse.m.o, community.m.o, blog.m.o, support.m.o, bugzilla.m.o, pontoon.m.o etc. with some potential user-submitted resources.

So if we consider ditching the ctfassets img src, I can try (RO) instead of specifically removing it to generally restrict the images to just self+GA and skip the defaults completely — to see if that triggers anything…

Reply:
I agree that explicit is better than wildcards here.

Grepping the source code...

* For `*.mozilla.net` I see we reference `assets.mozilla.net` and `mozorg.cdn.mozilla.net`.
* For `*.mozilla.com` I see `static.mozilla.com`. We link out to a lot of mozilla.com links but we don't need those in our CSP.
* For `*.mozilla.org` I don't see anything referencing this, so 'self' should suffice and this can be removed.